app/services/dwp/benefit_check_service.rb

```diff
@@ -1,10 +1,71 @@
 module DWP
   class BenefitCheckService
+    BENEFIT_CHECKER_NAMESPACE = 'https://lsc.gov.uk/benefitchecker/service/1.0/API_1.0_Check'.freeze
+    USE_MOCK = ActiveModel::Type::Boolean.new.cast(ENV.fetch('BC_USE_DEV_MOCK', nil))
+    REQUEST_TIMEOUT = 30.seconds
+
+    class ApiError < StandardError
+      include Nesty::NestedError
+    end
+
+    def initialize(applicant)
+      @applicant = applicant
+    end
+
     def self.call(applicant)
-      # For now we will just use the mock benefit check service
-      # and implement the full benefit check service functionality later
+      return MockBenefitCheckService.call(applicant) if USE_MOCK
+
+      new(applicant).call
+    end
+
+    def self.passporting_benefit?(applicant)
+      result = call(applicant)
+      return result if result.nil?
+
+      result[:benefit_checker_status].casecmp('yes').zero?
+    end
+
+    def call
+      soap_client.call(:check, message: benefit_checker_params).body[:benefit_checker_response]
+    rescue Savon::SOAPFault => e
+      Rails.logger.error(e)
+      Sentry.capture_exception(ApiError.new("HTTP #{e.http.code}, #{e.to_hash}"))
+      nil
+    rescue StandardError => e
+      Rails.logger.error(e)
+      Sentry.capture_exception(e)
+      nil
+    end
+
+    private
+
+    attr_reader :applicant
+
+    def benefit_checker_params
+      {
+        clientReference: applicant.crime_application_id,
```

Comment:
What are the DWP requirements for this `clientReference` value? Out of curiosity. Maybe we can use the UUID of the applicant record instead.